nodeJS/express/routing.md

```diff
@@ -0,0 +1,246 @@
+### Introduction
+
+In the previous lesson, we broke down a basic Express app and a request's journey through it. However, in a real app, we'd want to handle many different types of requests for different things.
+
+Routes essentially just match a request's HTTP verb (e.g. GET or POST) and URL path to the appropriate set of middleware functions - the controllers. We will explore controllers and middleware in more detail in the next lesson. For now, let's dive into how we can use routes.
+
+### Lesson overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Describe how routes are defined.
+- Match routes to specific HTTP verbs and paths.
+- Explain route parameters and query parameters.
+- Extract routes to their own routers.
```

Comment:
The second lesson overview can be skipped altogether as the first one encompasses it. Also, reworded the router overview (suggestions welcome):
```suggestion
- Describe how routes are defined.
- Explain route parameters and query parameters.
- Setup new routers on a path
```